altair/vegalite/v5/api.py

```diff
@@ -3813,29 +3813,14 @@ def add_selection(self, *params) -> Self:  # noqa: ANN002
         """'add_selection' is deprecated. Use 'add_params' instead."""
         return self.add_params(*params)
 
-    def interactive(  # noqa: C901
+    def interactive(
         self,
         name: str | None = None,
         bind_x: bool = True,
         bind_y: bool = True,
         tooltip: bool = True,
-        legend: bool
-        | Literal[
-            "color",
-            "fill",
-            "shape",
-            "stroke",
-            "opacity",
-            "fillOpacity",
-            "strokeOpacity",
-            "strokeWidth",
-            "strokeDash",
-            "angle",  # TODO Untested
-            "radius",  # TODO Untested
-            "radius2",  # TODO Untested
-            # "size",  # TODO Currently size is not working, renders empty legend
-        ] = True,
-    ) -> Self:
+        legend: bool | LegendChannel_T = False,
+    ) -> Chart:
```

Comment:
Using `Self` here is mainly useful when this is a base class.

I know this wasn't added in this PR, but using `Self` would cause conflicts in any generic "helper" functions like [`a8001da` (#3394)](https://github.com/vega/altair/pull/3394/commits/a8001da0245ad5d7515c2a90877c7b7cb2b91ce0#r1712993394)